src/enrichment/query/cosmos/utils.ts

```diff
@@ -24,3 +27,54 @@ export const findByKey = (
     ),
     TE.map(resp => pipe(resp.resource, O.fromNullable))
   );
+
+export const getQuery = (
+  containerName: string,
+  id: string,
+  versionFieldName: string,
+  versionFieldValue: string,
+  partitionKeyField: string,
+  partitionKeyValue: string
+): SqlQuerySpec => ({
+  parameters: [
+    { name: `@${versionFieldName}`, value: `${versionFieldValue}` },
+    { name: `@id`, value: `${id}` },
+    { name: `@${partitionKeyField}`, value: `${partitionKeyValue}` }
+  ],
+  query: `SELECT * FROM ${containerName} f WHERE  f.id = @id AND f.${versionFieldName} = @${versionFieldName} AND f.${partitionKeyField} = @${partitionKeyField}`
```

Comment:
just a little optimization: we should select TOP 1 and order by version field DESC